presto-native-execution/presto_cpp/main/PeriodicTaskManager.cpp

```diff
@@ -108,6 +108,8 @@ PeriodicTaskManager::PeriodicTaskManager(
       asyncDataCache_(asyncDataCache),
       arbitrator_(velox::memory::memoryManager()->arbitrator()),
       connectors_(connectors),
+      numStuckDriversToDetachWorker_(
```

Comment:
I think it would better to have a separate config, such as double type hw multiplier for detach. It could be either based on the hardware concurrency or a factor of driverNumCpuThreadsHwMultiplier? such as
numStuckDriversToDetachWorker_(hw concurrency * driverNumCpuThreadsHwMultiplier * stuckDriversToDetachWorkerFactor)?

Reply:
@xiaoxmeng 
So, you think we should have a separate config property for this?

What you describe looks like simple %:

pctStuckDriversToDetachWorker_
0 - never detach
1.0 - detach if all of them are blocked.
0.5 - half.
etc.

Or we can have a simple numStuckDriversToDetachWorker - a number of drivers, regardless of HW concurrency and number of cores.

Or we can have what is in this PR to not spawn yet another 'almost never used' config property.

I don't have a strong preference, just general strive for simplicity.

What would you prefer here?

Reply:
Chatted offline, will add plain number parameter.